pkg/monitor/cluster/cluster.go

```diff
@@ -60,10 +61,12 @@ type Monitor struct {
 		arodl *appsv1.DeploymentList
 	}
 
-	wg *sync.WaitGroup
+	wg                 *sync.WaitGroup
+	hiveClusterManager hive.ClusterManager
+	doc                *api.OpenShiftClusterDocument
 }
 
-func NewMonitor(log *logrus.Entry, restConfig *rest.Config, oc *api.OpenShiftCluster, m metrics.Emitter, hiveRestConfig *rest.Config, hourlyRun bool, wg *sync.WaitGroup) (*Monitor, error) {
+func NewMonitor(log *logrus.Entry, restConfig *rest.Config, oc *api.OpenShiftCluster, doc *api.OpenShiftClusterDocument, m metrics.Emitter, hiveRestConfig *rest.Config, hourlyRun bool, wg *sync.WaitGroup, hiveClusterManager hive.ClusterManager) (*Monitor, error) {
```

Comment:
It's a little strange to me that we have both `oc` and `doc` here (as `oc` should be a subproperty of `doc`), and `hiveRestConfig` and `hiveClusterManager`. 

I think there's an opportunity for us to deduplicate some of these dependencies, but that can be a follow-up refactor.

Reply:
We used the `doc` property here because it is utilized in the common function `GetClusterSync`, which is referenced in `pkg/frontend/admin_hive_syncset_resources.go`. Although this point was considered earlier, it led to code duplication in another area.